src/celeritas/em/distribution/WentzelDistribution.hh

```diff
@@ -0,0 +1,309 @@
+    // For incident electrons / positrons, theta_min = 0 always
+    const real_type w1 = 1 - cos_t1 + 2 * screen_coeff;
+    const real_type w2 = 1 - cos_t2 + 2 * screen_coeff;
+    const real_type cos_theta = clamp(
+        1 + 2 * screen_coeff - w1 * w2 / (w1 + uniform_sample(rng) * (w2 - w1)),
+        real_type{-1},
+        real_type{1});
+
+    // Calculate rejection for fake scattering
+    // TODO: Reference?
+    MottXsCalculator mott_xsec(element_data_, inc_energy_, inc_mass_);
+    const real_type form_factor
+        = calculate_form_factor(form_factor_coeff, cos_theta);
+    const real_type g_rej = mott_xsec(cos_theta) * ipow<2>(form_factor);
+
+    if (uniform_sample(rng) > g_rej)
+    {
+        return {0, 0, 1};
+    }
+
+    // Calculate scattered vector assuming azimuthal angle is isotropic
+    const real_type sin_theta = sqrt((1 - cos_theta) * (1 + cos_theta));
+    const real_type phi = 2 * celeritas::constants::pi * uniform_sample(rng);
+    return {sin_theta * cos(phi), sin_theta * sin(phi), cos_theta};
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculates the form factor based on the form factor model.
+ * TODO: Reference?
+ */
+CELER_FUNCTION real_type WentzelDistribution::calculate_form_factor(
+    real_type formf, real_type cos_t) const
+{
+    switch (data_.form_factor_type)
+    {
+        case NuclearFormFactorType::Flat: {
+            // In units MeV
+            const real_type ccoef = 0.00508;
+            const real_type x = sqrt(2 * inc_mom_sq() * (1 - cos_t)) * ccoef
+                                * 2;
+            return flat_form_factor(x)
+                   * flat_form_factor(
+                       x * 0.6
+                       * fastpow(value_as<Mass>(target_.nuclear_mass()),
+                                 static_cast<real_type>(1) / 3));
+        }
+        break;
+        case NuclearFormFactorType::Exponential:
+            return 1 / ipow<2>(1 + formf * (1 - cos_t));
+        case NuclearFormFactorType::Gaussian:
+            return exp(-2 * formf * (1 - cos_t));
+        default:
+            return 1;
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Flat form factor
+ * TODO: Reference?
+ */
+CELER_FUNCTION real_type WentzelDistribution::flat_form_factor(real_type x) const
+{
+    return 3 * (sin(x) - x * cos(x)) / ipow<3>(x);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Atomic number of the target isotope
+ */
+CELER_FUNCTION int WentzelDistribution::target_Z() const
+{
+    return target_.atomic_number().get();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Helper function to calculate the incident momentum squared.
+ */
+CELER_FUNCTION real_type WentzelDistribution::inc_mom_sq() const
+{
+    return inc_energy_ * (inc_energy_ + 2 * inc_mass_);
```

Comment:
Probably should not use the private data inside a helper function.

Reply:
I'm not sure I understand what you mean. Should I not be using a helper function for calculating momentum squared?